compiler-core/src/javascript.rs

```diff
@@ -345,18 +345,20 @@ impl<'a> Generator<'a> {
         imports: &mut Imports<'a>,
         package: &'a str,
         module: &'a str,
-        as_name: &'a Option<ImportName>,
+        as_name: &'a Option<(AssignName, SrcSpan)>,
         unqualified: &'a [UnqualifiedImport],
     ) {
-        let module_name = as_name
-            .as_ref()
-            .map(|n| SmolStr::as_str(&n.name))
-            .unwrap_or_else(|| {
-                module
-                    .split('/')
-                    .last()
-                    .expect("JavaScript generator could not identify imported module name.")
-            });
+        let name = module
+            .split('/')
+            .last()
+            .expect("JavaScript generator could not identify imported module name.");
```

Comment:
This now performs the work even if it is not needed- could make it into an anonymous perhaps?